mantidproject/mantid
# Workbench check for any C++ interfaces on startup

**Description of work.**

This work exposes the parts of `InterfaceManager` that deal with creating C++ interfaces to python so that they can be used in the Workbench.

**To test:**

This [patch](https://github.com/mantidproject/mantid/files/3134547/testing-interface-manager.txt) can be applied to the repository to add a testing C++ interface for Qt5. Apply the patch and build the code.
A `libTestingInterfaceManagerQt5.so` should be built into the `bin` directory.

Start the workbench and confirm that the new interface exists under a `Testing` submenu of `Interfaces` and that it starts.

Once finished remove the `libTestingInterfaceManagerQt5.so` from `bin`.

*This does not require release notes* because **it is a change to facilitate other work.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
